camel-core/src/main/java/org/apache/camel/language/bean/BeanLanguage.java

```diff
@@ -95,6 +95,12 @@ public Expression createExpression(String expression) {
             if (idx > 0) {
                 beanName = expression.substring(0, idx);
                 method = expression.substring(idx + 1);
+            } else {
+                int doubleColonIndex = expression.indexOf("::");
+                if (doubleColonIndex > 0) {
+                    beanName = expression.substring(0, doubleColonIndex);
+                    method = expression.substring(doubleColonIndex + 2);
+                }
```

Comment:
We should check first for `::`` then for `.` otherwise expressions like `my.own.Bean::function` would lead to error